index.html

```diff
@@ -6281,7 +6336,43 @@ <h3><dfn>Execute Async Script</dfn></h3>
       and data <var>result</var>.
 </ol>
 </section> <!-- /Execute Async Script -->
-</section> <!-- /Executing Script -->
+<section>
+ <h3><dfn>Pin Script</dfn></h3>
+
+ <table class="simple jsoncommand">
+  <tr>
+   <th>HTTP Method</th>
+   <th>URI Template</th>
+  </tr>
+  <tr>
+   <td>POST</td>
+   <td>/session/{<var>session id</var>}/execute/pin/{<var>name</var>}</td>
+  </tr>
+ </table>
+
+ <p>The <a>remote end steps</a> are:
+
+ <ol>
+  <li><p>Let <var>script</var> be the result of
+ <a>getting a property</a> named <var>script</var>
+ from the <var>parameters</var>.
+
+   <li><p>If <var>script</var> is <a>undefined</a> or is not a <a>String</a>,
+ return <a>error</a> with <a>error code</a> <a>invalid argument</a>.
+
+   <li><p>If the <a>current browsing context</a> is <a>no longer
+ open</a>, return <a>error</a> with <a>error code</a> <a>no such
+ window</a>.
+
+  <li><p>If <var>name</var> matches an <a>executable script name</a>
+ in the <a>list of known scripts</a> for the <a>current session</a>,
+ remove the script <var>name</var> from the <a>list of known scripts</a>.
+
+ <li><p>Add the script <var>name</var> to the <a>list of known scripts</a>
+ for the <a>current session</a>.
```

Comment:
This is fine, but not how you would implement this in actual code.

You would have a mutable `HashMap<String, String>` where the script value for the key _executable script name_ gets replaced. See also my above question about the duplication of the data structure.

Reply:
I modeled this after the wording in the Actions section where there was both a list and a table where the item in the list was the key value, isn't it essentially the same thing? How is that part implemented?

Reply:
I touch on this in https://github.com/w3c/webdriver/pull/1445#discussion_r334827587.

The effect is the same, but implementing a C-style hash map just seems a bit tedious and error prone.

Reply:
Based on comment elsewhere (that I missed when I posted this), I'll take a look at https://infra.spec.whatwg.org/#maps but I am still curious how it is different from Actions.

Reply:
addressed in https://github.com/w3c/webdriver/pull/1445#issuecomment-542264084